scripts/build.py

```diff
@@ -1102,10 +1102,22 @@ def build():
     parser = optparse.OptionParser()
     parser.add_option(
         '--prod_env', action='store_true', default=False, dest='prod_mode')
+    parser.add_option(
+        '--minify_third_party_libs_only', action='store_true', default=False,
+        dest='minify_third_party_libs_only')
     options = parser.parse_args()[0]
     # Regenerate /third_party/generated from scratch.
     safe_delete_directory_tree(THIRD_PARTY_GENERATED_DEV_DIR)
     build_third_party_libs(THIRD_PARTY_GENERATED_DEV_DIR)
+
+    # If minify_third_party_libs_only is set to True, skips the rest of the
+    # build process once third party libs are minified.
+    if options.minify_third_party_libs_only:
+        if not options.prod_mode:
+            raise Exception(
+                'minify_third_party_libs_only is set to true in non-prod mode.')
+        minify_third_party_libs(THIRD_PARTY_GENERATED_DEV_DIR)
+        return
```

Comment:
Instead of this, why not pull minify_third_party_libs() out of generate_build_directory() and do the former here instead? So:

```
if options.minify_third_party_libs_only and not options.prod_mode:
    raise Exception(...)

if options.prod_mode:
    minify_third_party_libs(...)
    if not options.minify_third_party_libs_only:
        generate_build_directory()
```

Might be clearer, because the story that's being told here is: check input args for validity and continue only if valid. Then, if in prod, do the always-required thing, and if necessary do the optional thing. I think this might be easier to understand since it's not obvious at the outset that generate_build_directory() includes minify_third_party_libs(), but what do you think?

Reply:
That is clearer indeed, thanks for the code pointer!